pkg/kwokctl/runtime/kind/cluster.go

```diff
@@ -379,10 +379,30 @@ func (c *Cluster) addKind(ctx context.Context, env *env) (err error) {
 	}
 
 	etcdComponentPatches := runtime.GetComponentPatches(env.kwokctlConfig, consts.ComponentEtcd)
+
 	kubeApiserverComponentPatches := runtime.GetComponentPatches(env.kwokctlConfig, consts.ComponentKubeApiserver)
 	kubeSchedulerComponentPatches := runtime.GetComponentPatches(env.kwokctlConfig, consts.ComponentKubeScheduler)
 	kubeControllerManagerComponentPatches := runtime.GetComponentPatches(env.kwokctlConfig, consts.ComponentKubeControllerManager)
 	kwokControllerComponentPatches := runtime.GetComponentPatches(env.kwokctlConfig, consts.ComponentKwokController)
+	for _, patch := range env.kwokctlConfig.ComponentsPatches {
+		switch patch.Name {
+		case consts.ComponentEtcd:
+			args := filterDuplicatedExtraArgs(etcdComponentPatches.ExtraArgs, patch.ExtraArgs)
+			etcdComponentPatches.ExtraArgs = args
+		case consts.ComponentKubeApiserver:
+			args := filterDuplicatedExtraArgs(kubeApiserverComponentPatches.ExtraArgs, patch.ExtraArgs)
+			kubeApiserverComponentPatches.ExtraArgs = args
+		case consts.ComponentKubeScheduler:
+			args := filterDuplicatedExtraArgs(kubeSchedulerComponentPatches.ExtraArgs, patch.ExtraArgs)
+			kubeSchedulerComponentPatches.ExtraArgs = args
+		case consts.ComponentKubeControllerManager:
+			args := filterDuplicatedExtraArgs(kubeControllerManagerComponentPatches.ExtraArgs, patch.ExtraArgs)
+			kubeControllerManagerComponentPatches.ExtraArgs = args
+		case consts.ComponentKwokController:
+			args := filterDuplicatedExtraArgs(kwokControllerComponentPatches.ExtraArgs, patch.ExtraArgs)
+			kwokControllerComponentPatches.ExtraArgs = args
+		}
+	}
```

Comment:
This will change the expected behavior.

We may need to design features that override existing args.

WDYT?

Reply:
filterDuplicatedExtraArgs overrides existing args using patch.ExtraArgs if args.Key is same.
So I look like that it's already designed. Am I misunderstanding ?

Reply:
What is indicated about expected behavior ?
Is it wrong that --extra-args patch the component if it's passed ? I think it's okay to change the expected behavior as a role of --extra-args. If am I misunderstanding, please let me know.

Reply:
> that this thing is an array.

I may know what you mean. But if the same key is passed, both is written, isn't is ? For example

`name1=key1=value1` and `name1=key1=value2`

is equal to

```yaml
- name1:
   extraArgs:
     - key: key1
        value: value1
    - key: key1
       value: value2
```

right ?

If so, which value is used by kwok ? And I think this configure may confuse to developer that look it. What do you think ?

Reply:
How about that ?

Reply:
I just noticed that Kind's configuration doesn't seem to allow keys to be duplicates? Maybe we should report the error and let the user decide?

Reply:
Sorry, I caught a cold, so late to respond.

I agree to file this problem to Kind community if a report is needed.
But I want to clarify why we allow keys to be duplicated.

If `name1=key1=value1` and `name1=key1=value2` are passed, what do you expect ?
And what do you suppose any usecase in kwok.

Reply:
I think it's different to pass duplicate elements to array and to process. So if we don't allow to pass same keys (as you referred to Kind), I think that we must make an error or filter same key.

Reply:
ping @wzshiming

Reply:
ohh, I'm sorry if I wasn't clear and you misunderstood me, what I meant was to add a warning message to this PR while de-duplicating it to let the user know it's being ignored.

Reply:
Oh, I'm sorry that I misunderstood. I could got it.
I added log if the key is duplicated.

Reply:
I selected the warning to let users know about information. But, is it better to report as an error than warning ?
What do you think ?